app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/SendGeoMessageViewModel.java

```diff
@@ -51,7 +61,16 @@ public String getFormattedPoint() {
   }
 
   public String[] getTypes() {
-    return mTypes;
+    return generateIconNames();
+  }
+
+  private String[] generateIconNames() {
+    String[] names = new String[mIcons.size()];
+    for (int i = 0; i < mIcons.size(); i++) {
```

Comment:
I think to myself, whether a foreach loop is nicer, but then u need to declare `names` as a `List` and add to it.
And in the return statement u use `Arrays.asList(names)`